src/taskgraph/transforms/from_deps.py

```diff
@@ -13,108 +13,93 @@
+class FetchEntry(Schema, rename=None):
+    """A fetch entry for an artifact."""
+
+    artifact: str
+    dest: Optional[str] = None
+
+
+class FromDepsConfig(Schema):
+    # Optional fields
+    # Limit dependencies to specified kinds (defaults to all kinds in
+    # `kind-dependencies`).
+    #
+    # The first kind in the list is the "primary" kind. The
+    # dependency of this kind will be used to derive the label
+    # and copy attributes (if `copy-attributes` is True).
+    kinds: Optional[List[str]] = None
+    # UPDATE ME AND DOCS
+    set_name: Optional[Union[str, bool, Dict[str, Any]]] = None
+    # Limit dependencies to tasks whose attributes match
+    # using :func:`~taskgraph.util.attributes.attrmatch`.
+    with_attributes: Optional[Dict[str, Union[List[Any], str]]] = None
+    # Group cross-kind dependencies using the given group-by
+    # function. One task will be created for each group. If not
+    # specified, the 'single' function will be used which creates
+    # a new task for each individual dependency.
+    group_by: Optional[Union[str, Dict[str, Any]]] = None
+    # If True, copy attributes from the dependency matching the
+    # first kind in the `kinds` list (whether specified explicitly
+    # or taken from `kind-dependencies`).
+    copy_attributes: Optional[bool] = None
+    # If true (the default), there must be only a single unique task
+    # for each kind in a dependency group. Setting this to false
+    # disables that requirement.
+    unique_kinds: Optional[bool] = None
+    # If present, a `fetches` entry will be added for each task
+    # dependency. Attributes of the upstream task may be used as
+    # substitution values in the `artifact` or `dest` values of the
+    # `fetches` entry.
+    fetches: Optional[Dict[str, List[Union[str, Dict[str, str]]]]] = None
+
+    def __post_init__(self):
+        # Validate set_name
+        if self.set_name is not None and self.set_name is not False:
+            if isinstance(self.set_name, str) and self.set_name not in SET_NAME_MAP:
+                raise msgspec.ValidationError(f"Invalid set-name: {self.set_name}")
+            elif isinstance(self.set_name, dict):
+                keys = list(self.set_name.keys())
+                if len(keys) != 1 or keys[0] not in SET_NAME_MAP:
+                    raise msgspec.ValidationError(
+                        f"Invalid set-name dict: {self.set_name}"
+                    )
+
+        # Validate group_by
+        if self.group_by is not None:
+            if isinstance(self.group_by, str) and self.group_by not in GROUP_BY_MAP:
+                raise msgspec.ValidationError(f"Invalid group-by: {self.group_by}")
+            elif isinstance(self.group_by, dict):
+                keys = list(self.group_by.keys())
+                if len(keys) != 1 or keys[0] not in GROUP_BY_MAP:
+                    raise msgspec.ValidationError(
+                        f"Invalid group-by dict: {self.group_by}"
+                    )
```

Comment:
I think you can get rid of this `__post_init__` function by creating a new type alias like you did for `priority` in `config.py`.